x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -393,33 +403,64 @@ protected void doRun() throws Exception {
                                         TimeValue.timeValueNanos(statsCurrentTimeNanosSupplier.getAsLong() - startTimeInNanos).millis()
                                     )
                                 );
+                            }));
+                        }
+                        enqueued = true;
+                    }
+                } catch (Exception e) {
+                    logger.warn(
+                        () -> new ParameterizedMessage("unable to open file [{}], cancelling prewarming for shard [{}]", fileName, shardId),
+                        e
+                    );
+                    if (queue.size() > 0) {
+                        try {
+                            ActionRunnable<Void> warmer;
+                            while ((warmer = queue.poll(0L, TimeUnit.MILLISECONDS)) != null) {
+                                warmer.onFailure(new IOException("Shard cache prewarming cancelled"));
                             }
+                        } catch (InterruptedException ie) {
+                            Thread.currentThread().interrupt();
```

Comment:
This just swallows the exception?

Reply:
I don't think we need to do more than logging here so I pushed b6b90c4e6c630972a8888dbadbc8222cf2ecc9bc